src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -360,7 +359,7 @@ protected void initProxy(ConnectionUrl connUrl) throws SQLException {
     this.initProxy(connUrl, ConnectionPluginManager::new);
   }
 
-  private void initProxy(
+  private synchronized void initProxy(
```

Comment:
Are we adding more synchronized keyword in this PR? Was this keyword here in the old code?

Reply:
Agree. It's a private function, no need for synchronized here.